lib/puppet/type/notify.rb

```diff
@@ -6,7 +6,7 @@ module Puppet
   Type.newtype(:notify) do
     @doc = "Sends an arbitrary message to the agent run-time log."
 
-    newproperty(:message) do
+    newproperty(:message, :idempotent => false) do
```

Comment:
👍 sounds like you have a staged plan of attack that's reasonable.

I'll see if I can come up with situations where the addition of `idempotent` to a property could break module authors - at the very least it might be worth a quick grep through the modules for collisions (seems unlikely to dig anything up).